tools/Packer/Program.cs

```diff
@@ -0,0 +1,157 @@
+            Directory.CreateDirectory(tempDirPath);
+            return tempDirPath;
+        }
+
+
+        private static void CloneRepoAndClean(string repoPath, string cloneUrl, List<string> pathsToRemove)
+        {
+            CloneRepo(cloneUrl, repoPath);
+
+            foreach (var path in pathsToRemove)
+            {
+                var combinedPath = Path.Combine(repoPath, path);
+                Console.WriteLine($"Removing: {combinedPath}");
+                if (File.Exists(combinedPath))
+                {
+                    File.Delete(combinedPath);
+                }
+
+                if (Directory.Exists(combinedPath))
+                {
+                    DeleteDirectoryRecursive(combinedPath);
+                }
+            }
+        }
+
+        private static void CreatePackage(string temporaryDirectory, ConfigModel config)
+        {
+            var packageName = Path.Combine(Environment.CurrentDirectory, $"{config.CloneDir}-{config.Version}.zip");
+            ZipFile.CreateFromDirectory(temporaryDirectory, packageName);
+        }
+
+        private static void CloneRepo(string cloneUrl, string repoPath)
+        {
+            var arguments = string.Join(" ", new string[] { "clone", cloneUrl, $"\"{repoPath}\"" });
+            
+            var info = new ProcessStartInfo("git", arguments)
+            {
+                CreateNoWindow = true,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                UseShellExecute = false,
+            };
+            
+            using (var process = Process.Start(info))
+            {
+                if (process == null)
+                {
+                    throw new Exception("Process failed to start");
+                }
+                
+                var processOutput = new StringBuilder();
+                
+                void OnReceived(object sender, DataReceivedEventArgs args)
+                {
+                    if (string.IsNullOrEmpty(args.Data))
+                    {
+                        return;
+                    }
+
+                    lock (processOutput)
+                    {
+                        processOutput.AppendLine(args.Data);
+                    }
+                }
+
+                process.OutputDataReceived += OnReceived;
+                process.ErrorDataReceived += OnReceived;
+
+                process.WaitForExit();
+
+                if (process.ExitCode != 0)
+                {
+                    Console.Error.WriteLine(processOutput.ToString());
+                    throw new Exception($"Process exited with a non-zero error code ({process.ExitCode})");
+                }
+                
+                Console.WriteLine(processOutput.ToString());
+            }
+        }
+
+
+        // Ensure that we can delete the files inside the directory by removing ReadOnly attribute from sub-files.
+        private static void DeleteDirectoryRecursive(string path)
```

Comment:
Rather than destructively deleting undesired files, why not write the logic for `ZipFile.CreateFromDirectory` and allow it exclude a set of paths (that is, use the `ZipArchive` APIs and extension methods.)

Reply:
That would probably be the safer method

Reply:
Implemented this instead!